src/cascadia/WindowsTerminal/WindowEmperor.cpp

```diff
@@ -73,9 +73,15 @@ bool WindowEmperor::HandleCommandlineArgs()
     _buildArgsFromCommandline(args);
     const auto cwd{ wil::GetCurrentDirectoryW<std::wstring>() };
 
-    // ALWAYS change the _real_ CWD of the Terminal to system32, so that we
-    // don't lock the directory we were spawned in.
-    LOG_IF_WIN32_BOOL_FALSE(SetCurrentDirectory(L"%SystemRoot%\\System32"));
+    {
+        // ALWAYS change the _real_ CWD of the Terminal to system32, so that we
+        // don't lock the directory we were spawned in.
+        std::wstring system32{};
+        if (SUCCEEDED_LOG(wil::GetSystemDirectoryW<std::wstring>(system32)))
+        {
+            LOG_IF_WIN32_BOOL_FALSE(SetCurrentDirectoryW(system32.c_str()));

```

Comment:
`system32` has no actual space at this point, since the string is empty. Using a `wchar_t system32[MAX_PATH];` should work and is the expected usage of the API.